libsolutil/Algorithms.h

```diff
@@ -68,6 +68,14 @@ class CycleDetector
 		return m_firstCycleVertex;
 	}
 
+	void reset()
```

Comment:
I would prefer assigning a newly created copy instead. Reset functions always have to make sure that they don't accidentally forget one of the members.